DOS-82: Schedule Component

AnasAmil1 opened the pull request.

Commit:
DOS-82: Schedule Component

Commit:
added gap on actions

jlhughes commented on src/foam/core/cron/CronSchedule.js:
```diff
@@ -158,7 +158,13 @@ foam.CLASS({
-      view: { class: 'foam.time.MonthOfYearView' },
+      view: function(_, X) {
+        return {
+          class: 'foam.u2.view.MultiChoiceDropdown',
```
This, and the others now default to selected/enabled.  Can you ensure the are not selected by default.

AnasAmil1 replied:
Hey, I don't really understand what you mean by they default to selected/enabled!

jlhughes replied:
Hithertoo, initially months of year, for example, are not 'selected'. 
With this change, when I select the down arrow for months of year, they all appear 'blue' - which indicates to me 'selected'.

jlhughes replied:
@sarthak-marwaha is reporting it appears correct to him, so I'll re-test.

AnasAmil1 replied:
I don't know why it's the case for you, but it's working fine for me. However, I changed it so that it extends from the Controller for Kevin's comment. That's probably irrelevant for your case, but can you check again, please, if it's fixed or not. Thank you

jlhughes replied:
Appears correct to me as well.

ajeetgill committed:
PT-FIXES: DOS-373 DOS-429 Improved property selector UX with multiple delimiter-separated values support

Users can now more easily select multiple properties using delimiter-separated input with better visual feedback and autocomplete suggestions.

- Extract common suggested field logic into PropertySuggestedField base class
- Create specialized ComparatorSuggestedField and PredicateSuggestedField classes
- Refactor ComparatorView and PredicateView to extend new base classes
- Add PropertyOptionCitationView for enhanced property dropdown UI
- Reduce code duplication across property selection components

ajeetgill committed:
updated listener

sarthak-marwaha committed:
cleanup

sarthak-marwaha committed:
more cleanup

Commit:
merged last changes

kgrgreer commented on src/foam/u2/view/MultiChoiceDropdown.js:
```diff
@@ -0,0 +1,175 @@
+        this.addClass()
+        .start()
+          .addClass(this.myClass('container'))
+            .startContext({ data: this })
```
You could extend foam.u2.Controller instead of foam.u2.View and then it would export 'this as data' instead and then you wouldn't need to startContext(). Then just add a 'data' property if you need one. Just FYI, I'm fine with either approach.

Commit:
Merge branch 'development' into fix/scheduleUI

sarthak-marwaha commented on src/foam/core/cron/CronSchedule.js:
```diff
@@ -207,7 +213,13 @@ foam.CLASS({
       order: 6,
       javaPreSet: 'if ( val != null ) { Arrays.sort(val); }',
-      view: { class: 'foam.u2.view.DayOfWeekView' },
+      view: function(_, X) {
```
We use the old views in NP, would it be possible to only change these in your repo? Or maybe offer a dropdown/segmented mode for selection when building the model

sarthak-marwaha commented on src/foam/u2/view/MultiChoiceDropdown.js:
```diff
@@ -0,0 +1,175 @@
+        .start()
+          .addClass(this.myClass('container'))
+            .startContext({ data: this })
+              .start(this.TOGGLE_OPEN).addClass(this.myClass('toggle-container'))
```
This should use the 'select' html element like choiceView does, not button

AnasAmil1 replied:
the select html element can't be styled inside the popover

sarthak-marwaha replied:
I meant that you use the UI from foam.u2.tag.Select so it looks like every other dropdown in FOAM but you maintain the on click behaviour so you can use your custom multiselect. Its the same thing we do in RichChoiceView

sarthak-marwaha commented on src/foam/u2/view/MultiChoiceDropdown.js:
```diff
@@ -0,0 +1,175 @@
+        window.addEventListener('mousedown', this.boundHandleClickOutside);
+      },
+
+      function destroy() {
```
What is this for?

Commit:
switched to controller

sarthak-marwaha commented on src/foam/core/reflow/ReactiveDetailView.js:
```diff
@@ -386,11 +386,16 @@ foam.CLASS({
       border-bottom: 1px solid $borderLight;
       padding: 8px 16px;
     }
+    ^ .property-schedule .foam-u2-detail-VerticalDetailView {
+      border: 1px solid $borderLight;
+      border-radius: 4px;
+    }
```
This change will fix the border in reflow but its broken everywhere else, can you fix it properly? If you change the dropdown to a select, you shouldnt need this anyway

AnasAmil1 replied:
this is not the border of select, this is of the view holding the peoperties, this one is only applied in reflow, because in the ReactiveSectionedDetailView, which is used only in reflow

sarthak-marwaha commented on src/foam/u2/view/MultiChoiceDropdown.js:
```diff
@@ -0,0 +1,173 @@
+        this.addClass()
+        .start()
+          .addClass(this.myClass('container'))
+            .start(this.TOGGLE_OPEN).addClass(this.myClass('toggle-container'))
```
This should use the 'select' html element like choiceView does, not button

AnasAmil1 replied:
The thing with select is that I give it choices, and this is a multiSelect dropdown, id don't want the standard HTML select.

sarthak-marwaha replied:
I mean that you use the foam.u2.tag.Select view so that this multiChoiceDropdown looks like every other select view and you get the added benefit of all the accessibility tags from Select copying over. Then you can change the onClick to call your action.
```
.start(foam.u2.tag.Select)
.on('click', this.toggleOpen)
```

sarthak-marwaha commented on src/foam/u2/view/MultiChoiceDropdown.js:
```diff
@@ -0,0 +1,173 @@
+        gap: 0.5rem;
+      }
+      ^ .foam-u2-view-MultiChoiceView-flexer > * {
+        flex: 1 0 16.666%; /* 6 columns: 100/6 */
```
Why do you do this, shouldnt this be handled by the numberOfColumns property?

AnasAmil1 replied:
That's because I changed it to flex, so flexer is not a grid anymore because i couldn't get the grid to look how i wanted without doing a lot of changes in the card view and other places

sarthak-marwaha replied:
Hmm, odd. That shouldnt be the case, you mind if i take a stab at it? I would like to have it working

AnasAmil1 replied:
Yes, sure, that would be helpful

sarthak-marwaha replied:
Fixed

sarthak-marwaha replied:
It now behaves as you were expecting, however the last one being the full width looks really weird imo, unless its a design requirement i liked the version where all the boxes are the same size better.

Commit:
removed destroy

abdelaziz-mahdy committed:
fix: remove labels from Mapping properties for cleaner UI

sarthak-marwaha committed:
Fix MulitChoiceView numberColumns

xuerongNanopay committed:
avoid sync contention if parser already exists

abdelaziz-mahdy committed:
feat: add fromString method for Float type conversion

ajeetgill committed:
fixed space formatting

ajeetgill committed:
PT-Fixes DOS-376 allows second level menu items to have Icons
AND add gaps between elements which were looked stuck together

kgrgreer committed:
PT-FIXES: DOS-417 DAOPrompt refreshes rowCount and display when DAO changes.

sarthak-marwaha commented on src/foam/u2/view/MultiChoiceDropdown.js:
```diff
@@ -0,0 +1,160 @@
+            .end()
+            .start()
+              .add(this.dynamic(function(isOpen) {
+                if (isOpen) {
```
Can you switch this to use OverlayDropdown, you will get all the open/close logic for free and it behaves better when your dropdown elements are too close to the edge

AnasAmil1 replied:
Done

kgrgreer committed:
Reflow commands leaving off 'DAO' as daoKey.

abdelaziz-mahdy committed:
fix: simplify expression evaluation by removing unnecessary scope

jlhughes committed:
DAO Compaction - refactor medusa compaction for single dao compaction

Compaction dumps an MDAO to a new journal, writing each record in
entirety, replacing the original journal which may have a long history
of partial entries as a result of object updates.
Journal replay parsing time is relative to lines parsed, the less lines,
the less time replay takes.
The process supports faceted Sinks to control the compaction of
individual models.

jlhughes committed:
script - update comments

jlhughes committed:
Compaction - update documentation

jlhughes committed:
Compaction - add additional documentation regarding archiving

ajeetgill committed:
Added PropertyOptionCitationView (shows both label + name) making GUI readble for dev&nondev audience, & minor ux improvements for end-user

kgrgreer committed:
PT-FIXES: DOS-360 Fix nested GroupBys Browse functionality setting the wrong primary key on the generated model and loosing values. Also add a FLOAT test to verify correctness.

jlhughes committed:
documenation - add discussion of FixedSizeDAO

ajeetgill committed:
Update src/foam/core/reflow/PropertyOptionCitationView.js

Co-authored-by: sarthak <81172969+sarthak-marwaha@users.noreply.github.com>

sarthak-marwaha committed:
Merge Titled and RichTitled arrayView

kgrgreer committed:
Fix typo in reflow cmds.jrl which broke upload.

ajeetgill committed:
removed excess css, and added "..." for long propertyName (label still show fully and wrap text to next line)

ajeetgill committed:
Update src/foam/core/reflow/PropertyOptionCitationView.js

Co-authored-by: sarthak <81172969+sarthak-marwaha@users.noreply.github.com>

ajeetgill committed:
removed code coz 'symbol' never used

kgrgreer committed:
Add toolTip for Reflow.fullScreen toggle.

sarthak-marwaha committed:
Fix some spacing

sarthak-marwaha committed:
fix grid gap

abdelaziz-mahdy committed:
feat: add FOAM3 Cheat Sheet with comprehensive model and property definitions

jlhughes committed:
Script - implement Authorizable to facilitate permission based Script access and execution

jlhughes committed:
FailedLoginAuthService - Log login failures

abdelaziz-mahdy committed:
feat: enhance FOAM3 Cheat Sheet with expression behavior and reactivity examples

abdelaziz-mahdy committed:
fix: restore preLoad call in flow loading process

jlhughes committed:
LoginAttempt - record failed login attempts to a DAO, providing for other rule based actions

kgrgreer committed:
Fix syntax error in flows.jrl.

jlhughes committed:
build - softlinks, test arguments, appRoot for tests and benchmarks

emptyDir,rmdir - if directory is symbolic link, remove contents rather
than directory itself (so not to destroy links to ram disks).
testArgs - add support to pass arguments to test cases and benchmarks
tests/benchmarks - honour appRoot setting.

VesnaSUG committed:
Adjusted formatting

kgrgreer committed:
Change label in TitledArrayView from 'Add New' to just 'Add'.

kgrgreer committed:
Make upload command linkable false.

abdelaziz-mahdy committed:
fix empty fields when uploading

kgrgreer committed:
Revert previous change to Upload command.

kgrgreer committed:
Fix NPE in Reflow clear menu.

kgrgreer committed:
PT-FIXES: DOS-417 skip redraws multiple copies of result.

kgrgreer committed:
PT-FIXES: DOS-479 DAOPrompt skip works better on low bandwidth connections.

kgrgreer committed:
PT-FIXES: DOS-368 Fix top-level tree node name to 'flow' to make scripting easier.

jlhughes committed:
build - add RAM Disk creation with symbolic link to build directory

jlhughes committed:
RAMDiskTooling - reduce default disk size to 500 - enough for known large projects

kgrgreer committed:
PT-FIXES: DOS-480 DAOPrompt Scroll limit max range not set correctly.

Commit:
swicthed to dropdown

Commit:
fixed conflicts

sarthak-marwaha commented:
You might want to rebase this branch onto dev, was this a bad merge?
